src/numeric_index.c

```diff
@@ -147,10 +124,57 @@ size_t NumericRange_Add(NumericRange *n, t_docId docId, double value, int checkC
   return size;
 }
 
-double NumericRange_Split(NumericRange *n, NumericRangeNode **lp, NumericRangeNode **rp,
+// static double NumericRange_EstimateMedian(NumericRange *n) {
+//   size_t numSamples = 10 * (31 - __builtin_clz(n->entries->numDocs));
+//   // Sample the range uniformly
+//   // Pick values with IR_SkipTo
+//   // Get the median value of the samples
+// }
+
+typedef union {
+  void *ptr;
+  double val;
+} value_holder;
+static_assert(sizeof(void*) >= sizeof(double));
+
+static int cmp_values(const void *p1, const void *p2, const void *udata) {
+  (void)udata;
+  const value_holder v1 = { .ptr = (void*)p1 };
+  const value_holder v2 = { .ptr = (void*)p2 };
+  if (v1.val < v2.val) return -1;
+  return (v1.val > v2.val) ? 1 : 0;
+}
+static double NumericRange_GetMedian(IndexReader *ir) {
+  size_t media_idx = ir->idx->numDocs / 2;
```

Comment:
is it accurate enough to split in the middle?

Reply:
Are you referring to taking the average of the 2 medians in case we have an even number of docs?
I think this is good enough. An easy estimation for 50-50 split